src/TensorOneD.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <math.h>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+namespace geopm
+{
+    TensorOneD::TensorOneD()
+    {
+    }
+
+    TensorOneD::TensorOneD(int dim)
+    {
+        set_dim(dim);
+    }
+
+    TensorOneD::TensorOneD(const TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+    }
+
+    TensorOneD::TensorOneD(json11::Json input)
+    {
+        set_dim(input.array_items().size());
+        for (int idx = 0; idx < (int)m_vec.size(); idx++) {
+            m_vec[idx] = input[idx].number_value();
+        }
+    }
+
+    int TensorOneD::get_dim() const
+        {
+            return m_vec.size();
+        }
+
+    void TensorOneD::set_dim(int dim)
+        {
+            m_vec.resize(dim);
+        }
+
+    TensorOneD TensorOneD::operator+(const TensorOneD& other)
+        {
+            if (get_dim() != other.get_dim()) {
+                throw geopm::Exception("Adding vectors of mismatched dimensions.",
+                        GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+
+            TensorOneD rval(m_vec.size());
+            std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::plus<float>());
+
+            return rval;
+        }
+
+    TensorOneD TensorOneD::operator-(const TensorOneD& other)
+        {
+            if (get_dim() != other.get_dim()) {
+                throw geopm::Exception("Subtracting vectors of mismatched dimensions.",
+                        GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+
+            TensorOneD rval(m_vec.size());
+            std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::minus<float>());
+
+            return rval;
+        }
+
+
+    float TensorOneD::operator*(const TensorOneD& other)
+        {
+            if (get_dim() != other.get_dim()) {
+                throw geopm::Exception("Inner product of vectors of mismatched dimensions.",
+                        GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+
+            return std::inner_product(m_vec.begin(), m_vec.end(), other.m_vec.begin(), 0);
+        }
+
+    TensorOneD& TensorOneD::operator=(const TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+        return *this;
+    }
+
+    float& TensorOneD::operator[] (int idx)
```

Comment:
Sorry, I should've noticed this before:

A similar comment about leveraging the vector's ``size_type`` applies here as well (instead of using int). See: https://cplusplus.com/reference/vector/vector/operator[]/

The same applies to other places in the code (e.g. the specialized constructor that takes the dim).